app/core/database_service.py

```
@@ -0,0 +1,558 @@
+            
+            if operation == "select":
+                rows_affected = len(result.all()) if hasattr(result, 'all') else 1
+            else:
+                rows_affected = result.rowcount if hasattr(result, 'rowcount') else 0
+            
+            success = True
+            self.connection_pool_stats["successful_queries"] += 1
+            return result
+            
+        except (IntegrityError, DataError) as e:
+            error_message = f"Data integrity error: {str(e)}"
```

Ruff (0.13.1)

Use explicit conversion flag

Replace with conversion flag

(RUF010)